### ISSUE TITLE
Bug (ish): `tagName: ElementType` is too wide

### ISSUE THREAD

**Issue by psychedelicious** [bvaughn/react-resizable-panels#251]
## Problem

Exported components accept a `tagName` prop to determine which element type to use for its DOM elements.

This is currently typed as `ElementType`, but this is a bit too wide as it allows any `Element`. For example, this allows tags like `svg`, which don't have the attributes needed for the library to work and cause runtime errors. I think only `HTMLElement`s work.

It's hard to imagine a situation in which this would ever actually cause problems, but it's nice to have correct types of course.

## Proposal 1

Narrow the type to of `tagName` so that it only accepts tags for `HTMLElement`s. It looks like the way to do this is `keyof HTMLElementTagNameMap`:

```ts
export type PanelProps = Omit<HTMLAttributes<keyof HTMLElementTagNameMap>, "id" | "onResize"> &
  PropsWithChildren<{
    ...
    tagName?: keyof HTMLElementTagNameMap;
  }>;
```

I was surprised to not find a direct type e.g. `HTMLElementType`. Also, `JSX.IntrisicElements` includes non-HTML elements, so that doesn't help.

## Proposal 2

*Really* narrow the type down to just a handful of specific tags - like `div`, `span`, etc. Feels a bit restrictive, and not future-proofed.

## Proposal 3

Just update the docs with a warning.

**Comment by bvaughn** [comment]
I don't feel strongly about this one. If you want to submit a PR, I'll look it over. Probably won't do anything myself for this though.

**Comment by bvaughn** [comment]
The PR you submitted is an improvement (thanks!) although I'm wondering about the distinction between `HTMLAttributes` and `AllHTMLAttributes`. It looks like some attributes (like `value`  for `HTMLDivElement`) are excluded from `HTMLAttributes` whereas others (like `defaultValue`) are included. `AllHTMLAttributes` includes both, but I'm not sure how to refine it based on the specified tag name.

For example:
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDGMA0cDecCCANoQBIAqAsgDL4wxTABGArjEgM7bnW31OsdsaKKjYBRQkhBIAdvAC+cAGZQIIOACIR6GBoDcAKAMwAnmCRxuVMigDmAORTS4AXjgBrJCYhLLlKhJSsjA2Dk5IFChghgZKzDIYwBAycABCqigAJgDCapAywQA8oY7OSAAebDKZ7H7UJeEAfAAUOAZwcDB2pUiY7XAAdEMi7DAG8gBcBMRWvAwsbOzF3U1wAGS4-V1h0lMN0uMAlJsdIjDMUCnCokiB0nLN2z3YIzCHhvJGcQkwSSmOUFUAHdcuBkkV9hYKlUanVrCtpC02h0nuE+h0hgNXuMprM6PMBEtIY11idOgikHsKUcyWcLldtOJJPcYI8KS8OG8PkZkXBCukIFlQflguSdkgXBpgDIwKwNJ1gDBJJLFro4AA3FCEZgSnAARgATABmRQAeka4wMvMKAOBwvBcjFPUl0tlat+SolGlV8s12t1hpNcHNlqt-VNprgAAEYOwALQVcwYBOA6D9fkZHJ5B3wVHSSWZYDq+Ue5Xezm+rU6lz641mi2fMMdCPR2MJ8pJmAp1RQdO2iAg7MFR15r2F4sKz0qisaqsBuvBhsGIA

Maybe this is a limitation of TypeScript or maybe just a gap in my knowledge. Going to poke at it some more before making a release.

**Comment by bvaughn** [comment]
Maybe this is not worth the effort and I should just stick with the incremental improvement of your change.

Published react-resizable-panels@1.0.7

**Comment by psychedelicious** [comment]
I played around a bit but it's beyond me. If a user needs this degree of type safety they will have a good puzzle.

Thanks for the review and extra investigation.
